src/Circle.js

```diff
@@ -0,0 +1,95 @@
+import React, { useRef, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const DIRECTIONS = {
+    CLOCKWISE: 'clockwise',
+    ANTI_CLOCKWISE: 'antiClockwise',
+    BOTH_SIDES: 'bothSides',
+};
+
+const getSvgClassName = (strokePercentage, direction, className) => classNames(
+    'react-circle_svg',
+    `react-circle_${direction}`,
+    {
+        'react-circle_noStroke': strokePercentage === 0,
+    },
+    className,
+);
+
+const Circle = ({ className, direction, strokeWidth, strokePercentage, onTransitionEnd }) => {
+    const svgRef = useRef(null);
+    const [circleState, setCircleState] = useState({});
```

Comment:
You are initializing `circleState` with an empty object but you can initialize with no values: `useState()`.
What you did is not wrong, but as far as I understand, it seems unnecessary 😬

Reply:
It may be a result of past implementations if circleState was ever accessed before the useEffect call. Since this is not the case anymore, I'll remove it.